File: javascript/fibonacci.js
function fibonacci(num) {
  let arr = [0, 1]

  for (let i = 2; i <= num; i++) {
    arr.push(arr[i-1] + arr[i-2])
  }

  return arr[num]
}

// Eva's pseudocode/explanation
  // Find the nth element
    // starts with 0, 1
    // 0, 1, 1, 2, 3, 5, 8
  // After that, every value is the sum of the two values preceding it
  // attn: zero-indexing



if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 0");
  console.log("=>", fibonacci(0));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", fibonacci(2));

  console.log("");

  console.log("Expecting: 55");
  console.log("=>", fibonacci(10));
}

module.exports = fibonacci;

// Please add your pseudocode to this file
// And a written explanation of your solution

File: ruby/fibonacci.rb
def fibonacci(num)
  # type your code in here
  x, y = [0, 1]

  num.times do
    x, y = y, x + y
  end
  
  x
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 0"
  puts "=>", fibonacci(0)

  puts

  puts "Expecting: 1"
  puts "=>", fibonacci(2)

  puts

  puts "Expecting: 55"
  puts "=>", fibonacci(10)

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution


# Eva's first attempt:
  # arr << (arr[num-1] + arr[num-2])

  # DRAFT 1

  # def fibonacci(num)
  #   # type your code in here
  #   first_num, second_num = [0, 1]
  
  #   num.times do
  #     first_num, second_num = second_num, first_num + second_num
  #   end
  
  #   first_num
  # end
  